calculator_test.go

```diff
@@ -2,23 +2,200 @@ package calculator_test
 
 import (
 	"calculator"
+	"fmt"
+	"math/rand"
 	"testing"
 )
 
+type testCase struct {
```

Comment:
Since test case structs tend to be test-specific, we often don't give them a name at all, but just use an _anonymous struct type_:

```go
testCases := []struct{
	a, b float64
	...
}{
	{a:2, b:2, ... },
}
```

Reply:
Would this be declared in each of the Test* functions and remove the need for the generic types at the top?

Reply:
Yes, that's a fairly standard pattern for tests.

Reply:
Cool. cf0cf63 addresses this.